### ISSUE TITLE
Julia 0.4 support?

### ISSUE THREAD

**Issue by schmrlng** [Keno/Cxx.jl#239]
Is Julia 0.4 intended to be supported going forward? Currently ```Pkg.build("Cxx")``` fails with this Make.user
```
override LLVM_ASSERTIONS=1
override BUILD_LLVM_CLANG=1
override USE_LLVM_SHLIB=1
override LLVM_VER=3.7.1
```
on the release-0.4 branch. See this [gist](https://gist.github.com/schmrlng/faaa723de9206bb1a0a3).

Alternatively, is there a particular commit on the current julia master (0.5) we should target where Cxx.jl is known to be working as intended? Aside from https://github.com/Keno/Cxx.jl/issues/236 which still stands, the current julia master gives me this behaviour (it seems that a lot of the behind-the-scenes JIT plumbing is being rewritten these days; this branch https://github.com/JuliaLang/julia/pull/15609 suggested as a potential fix for https://github.com/Keno/Cxx.jl/issues/236 doesn't work either):
```
   _       _ _(_)_     |  A fresh approach to technical computing
  (_)     | (_) (_)    |  Documentation: http://docs.julialang.org
   _ _   _| |_  __ _   |  Type "?help" for help.
  | | | | | | |/ _` |  |
  | | |_| | | | (_| |  |  Version 0.5.0-dev+3293 (2016-03-28 13:36 UTC)
 _/ |\__'_|_|_|\__'_|  |  Commit 2103b17 (0 days old master)
|__/                   |  x86_64-linux-gnu

julia> using Cxx

julia> cxx"""#include <iostream>"""
julia: /home/schmrlng/code/oss/julia-0.5/src/jitlayers.cpp:358: {anonymous}::JuliaOJIT::ModuleHandleT {anonymous}::JuliaOJIT::addModule(std::unique_ptr<llvm::Module>): Assertion `F->isIntrinsic() || findUnmangledSymbol(F->getName()) || SectionMemoryManager::getSymbolAddressInProcess(F->getName())' failed.
[1]    2501 abort (core dumped)  julia
```

**Comment by Keno** [comment]
Julia 0.4 is not supported. Julia 0.5 is technically supported, but was recently broken on julia master and I haven't gotten around to fixing it yet (plus there's another issue which requires jb/linear to be merged). I'll try to get everything in working order again once jb/linear is merged. Until then, I'd try some commit in early January.

**Comment by schmrlng** [comment]
Cool, thanks for the clarification!
